### PR TITLE
Added unit tests for `storage` package

### PR DESCRIPTION
**Pull request by iulianpascalau**
## Reasoning behind the pull request
- the `storage` package had a low code coverage percentage
  
## Proposed changes
- minor refactoring in storage package
- added unit tests

## Testing procedure
- standard system test

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: storage/mock/storedDataFactoryStub.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package mock
```
Comment (AdoAdoAdo) [inline storage/mock/storedDataFactoryStub.go:1]: Please move this somewhere in the testscommon instead.

File: storage/pathmanager/pathManager_test.go [hunk 2]
```diff
@@ -11,66 +12,82 @@ func TestNewPathManager_EmptyPruningPathTemplateShouldErr(t *testing.T) {
 	t.Parallel()
 
 	pm, err := pathmanager.NewPathManager("", "shard_[S]/[I]", "db")
-	assert.Nil(t, pm)
+	assert.True(t, check.IfNil(pm))
```
Comment (AdoAdoAdo) [inline storage/pathmanager/pathManager_test.go:15]: please revert to assert.Nil to all errors
assert.True is not good for nil checks as it requires more time to figure out the issue looking at the tests logs.

File: storage/pathmanager/pathManager_test.go [hunk 3]
```diff
@@ -11,66 +12,82 @@ func TestNewPathManager_EmptyPruningPathTemplateShouldErr(t *testing.T) {
...
 	assert.Equal(t, pathmanager.ErrInvalidStaticPathTemplate, err)
 }
 
 func TestNewPathManager_OkValsShouldWork(t *testing.T) {
 	t.Parallel()
 
 	pm, err := pathmanager.NewPathManager("epoch_[E]/shard_[S]/[I]", "shard_[S]/[I]", "db")
-	assert.NotNil(t, pm)
+	assert.False(t, check.IfNil(pm))
```
Comment (AdoAdoAdo) [inline storage/pathmanager/pathManager_test.go:79]: please revert to NotNil check

**Review by AdoAdoAdo — CHANGES_REQUESTED** [review]

File: storage/disabled/errorDisabledPersister.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+// Remove returns error
+func (disabled *errorDisabledPersister) Remove(_ []byte) error {
+	return fmt.Errorf("disabledPersister.Remove")
+}
+
+// Destroy does nothing
+func (disabled *errorDisabledPersister) Destroy() error {
+	return fmt.Errorf("disabledPersister.Destroy")
+}
```
Comment (ssd04) [inline storage/disabled/errorDisabledPersister.go:43]:
```suggestion
// Destroy returns error
func (disabled *errorDisabledPersister) Destroy() error {
	return fmt.Errorf("disabledPersister.Destroy")
}
```

File: storage/disabled/errorDisabledPersister_test.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+	disabled := NewErrorDisabledPersister()
+	t.Run("nil handler should not panic", func(t *testing.T) {
+		t.Parallel()
+
+		assert.NotPanics(t, func() {
+			disabled.RangeKeys(nil)
+		})
+	})
+	t.Run("handler should not be called", func(t *testing.T) {
+		disabled.RangeKeys(func(key []byte, val []byte) bool {
```
Comment (ssd04) [inline storage/disabled/errorDisabledPersister_test.go:86]: we can add parallel run here also

File: storage/mock/storedDataFactoryStub.go (hunk 1 above)
Comment (iulianpascalau) [reply storage/mock/storedDataFactoryStub.go:1]: moved in testscommon/storage package

File: storage/pathmanager/pathManager_test.go (hunk 2 above)
Comment (iulianpascalau) [reply storage/pathmanager/pathManager_test.go:15]: reworked all tests from `storage` package. Left `check.IfNil(pm)` only when the tested function returned an interface.

File: storage/disabled/errorDisabledPersister.go (hunk 4 above)
Comment (iulianpascalau) [reply storage/disabled/errorDisabledPersister.go:43]: changed

File: storage/pathmanager/pathManager_test.go (hunk 3 above)
Comment (iulianpascalau) [reply storage/pathmanager/pathManager_test.go]: reworked all tests from storage package. Left check.IfNil(pm) only when the tested function returned an interface.

**Review by ssd04 — APPROVED** [review]

File: storage/pruning/triePruningStorer_test.go [hunk 6]
```diff
@@ -350,3 +349,45 @@ func TestTriePruningStorer_KeepMoreDbsOpenIfNecessary(t *testing.T) {
...
+	t.Run("no active db should error", func(t *testing.T) {
+		t.Parallel()
+
+		tps, _ := pruning.NewTriePruningStorer(args)
+		latestEpoch, err := tps.GetLatestStorageEpoch()
+		expectedErrString := "there are no active persisters"
+		assert.Equal(t, expectedErrString, err.Error())
+		assert.Zero(t, latestEpoch)
+	})
+	t.Run("no active db should error", func(t *testing.T) {
```
Comment (AdoAdoAdo) [inline storage/pruning/triePruningStorer_test.go:373]: wrong message -> should work?

**Review by AdoAdoAdo — CHANGES_REQUESTED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed 
@@                            Log scanner                                  @@
##                            add-unit-tests-for-storage                   ##
================================================================================
- Known Warnings              8
- New Warnings                1
- Known Errors                0
- New Errors                  0
- Panics                      0
================================================================================
- block hash does not match                  3910
- wrong nonce in block                       1425
- miniblocks does not match                  0
- num miniblocks does not match              0
- miniblock hash does not match              0
- block bodies does not match                0
- receipts hash missmatch                    0
================================================================================
-  No jailed nodes on the thestnet
================================================================================

**Pull request merged**